### DISCUSSION TITLE
After creating a custom icon using 'npx ignite cli generate app icon all -- skip source quality validation', an APK cannot be built.

### DISCUSSION THREAD

**Discussion by bihongbin3027** [infinitered/ignite#2542] [General]
> Task :app:mergeReleaseResources FAILED
ERROR:[raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_300light] D:\sxy-web\SpiderSMS\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_300light.ttf [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_300light] D:\sxy-web\SpiderSMS\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_300light.ttf: Resource and asset merger: Duplicate resources
ERROR:[raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_400regular] D:\sxy-web\SpiderSMS\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_400regular.ttf [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_400regular] D:\sxy-web\SpiderSMS\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_400regular.ttf: Resource and asset merger: Duplicate resources
ERROR:[raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_500medium] D:\sxy-web\SpiderSMS\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_500medium.ttf [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_500medium] D:\sxy-web\SpiderSMS\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_500medium.ttf: Resource and asset merger: Duplicate resources
ERROR:[raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_600semibold] D:\sxy-web\SpiderSMS\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_600semibold.ttf [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_600semibold] D:\sxy-web\SpiderSMS\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_600semibold.ttf: Resource and asset merger: Duplicate resources
ERROR:[raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_700bold] D:\sxy-web\SpiderSMS\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_700bold.ttf [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_700bold] D:\sxy-web\SpiderSMS\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_700bold.ttf: Resource and asset merger: Duplicate resources

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:mergeReleaseResources'.
> [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_300light] D:\sxy-web\SpiderSMS\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_300light.ttf      [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_300light] D:\sxy-web\SpiderSMS\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_300light.ttf: Error: Duplicate resources
  [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_400regular] D:\sxy-web\SpiderSMS\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_400regular.ttf  [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_400regular] D:\sxy-web\SpiderSMS\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_400regular.ttf: Error: Duplicate resources
  [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_500medium] D:\sxy-web\SpiderSMS\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_500medium.ttf    [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_500medium] D:\sxy-web\SpiderSMS\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_500medium.ttf: Error: Duplicate resources
  [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_600semibold] D:\sxy-web\SpiderSMS\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_600semibold.ttf        [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_600semibold] D:\sxy-web\SpiderSMS\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_600semibold.ttf: Error: Duplicate resources
  [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_700bold] D:\sxy-web\SpiderSMS\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_700bold.ttf        [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_700bold] D:\sxy-web\SpiderSMS\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_700bold.ttf: Error: Duplicate resources

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 18s
48 actionable tasks: 6 executed, 42 up-to-date
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by bihongbin3027** [comment]
Tested and used normally in the debug environment, I want to build APK, but it cannot succeed.

Yarn bundle: Android

Yarn release: Android

**Reply by frankcalise** [reply]
Hi, sorry for your troubles!

Can you confirm you have built a release APK prior to this icon generation?

Also would you mind posting the output of `npx ignite-cli doctor`?

**Comment by bihongbin3027** [comment]
Hi, I thought it was a problem with using the "npx ignite cli generate app icon all -- skip source quality validation" command before producing the app. Just now I used 'npx' ignite-cli@latest The new PizzaApp has recreated the project, but still cannot package the APK, the same problem as the one above: 'Resource and asset merge: Duplicate resources'.

The following are the results of the 'npx ignite cli doctor':

$npx ignite cli doctor

Npm WARN exec The following package was not found and will be installed: ignite-cli@8.10.1

System

Platform win32

Arch x64

CPU 12 cores Intel (R) Core (TM) i5-10500 CPU @ 3.10GHz

Directory SpiderSMS D: sy web SpiderSMS



JavaScript (and globally installed packages)

Node 18.17.0 C: Program Files nodejs node EXE

Npm 10.1.0 C: Program Files nodejs npm CMD

Corepack 0.18.0

NPM 10.1.0

Pnpm 8.7.5

React native cli 2.0.1

Yarn 1.22.19

Yarn 1.22.19 C: Program Files nodejs yarn CMD

Pnpm 8.7.5 C: Program Files nodejs pnpm CMD

<no pnpm global package information available>

Expo48.0.21 managed



Ignite

IGNITE-CLI 8.10.1 C: Users sxy AppData Local npm cache\_ Npx e31027f3785124a8 node_ Modules . Bin ignite.CMD

IGNITE src build C: Users sxy AppData Local npm cache\_ Npx e31027f3785124a8 node_ Modules ignite cli build



Android

Java 11.0.20 C: Program Files Java jdk-11 bin java EXE

Android home - C: Users sxy AppData Local Android Sdk



Tools

Git git version 2.40.0. windows-1 C: Program Files Git mingw64 bin git EXE

**Comment by bihongbin3027** [comment]
yarn release:android

> Task :app:mergeReleaseResources FAILED
ERROR:[raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_300light] D:\sxy-web\PizzaApp\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_300light.ttf [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_300light] D:\sxy-web\PizzaApp\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_300light.ttf: Resource and asset merger: Duplicate resources
ERROR:[raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_400regular] D:\sxy-web\PizzaApp\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_400regular.ttf [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_400regular] D:\sxy-web\PizzaApp\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_400regular.ttf: Resource and asset merger: Duplicate resources
ERROR:[raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_500medium] D:\sxy-web\PizzaApp\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_500medium.ttf [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_500medium] D:\sxy-web\PizzaApp\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_500medium.ttf: Resource and asset merger: Duplicate resources
ERROR:[raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_600semibold] D:\sxy-web\PizzaApp\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_600semibold.ttf [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_600semibold] D:\sxy-web\PizzaApp\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_600semibold.ttf: Resource and asset merger: Duplicate resources
ERROR:[raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_700bold] D:\sxy-web\PizzaApp\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_700bold.ttf [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_700bold] D:\sxy-web\PizzaApp\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_700bold.ttf: Resource and asset merger: Duplicate resources

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:mergeReleaseResources'.
> [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_300light] D:\sxy-web\PizzaApp\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_300light.ttf    [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_300light] D:\sxy-web\PizzaApp\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_300light.ttf: Error: Duplicate resources
  [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_400regular] D:\sxy-web\PizzaApp\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_400regular.ttf        [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_400regular] D:\sxy-web\PizzaApp\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_400regular.ttf: Error: Duplicate resources
  [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_500medium] D:\sxy-web\PizzaApp\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_500medium.ttf  [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_500medium] D:\sxy-web\PizzaApp\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_500medium.ttf: Error: Duplicate resources
  [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_600semibold] D:\sxy-web\PizzaApp\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_600semibold.ttf      [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_600semibold] D:\sxy-web\PizzaApp\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_600semibold.ttf: Error: Duplicate resources
  [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_700bold] D:\sxy-web\PizzaApp\android\app\src\main\res\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_700bold.ttf      [raw/node_modules_expogooglefonts_spacegrotesk_spacegrotesk_700bold] D:\sxy-web\PizzaApp\android\app\build\generated\res\react\release\raw\node_modules_expogooglefonts_spacegrotesk_spacegrotesk_700bold.ttf: Error: Duplicate resources

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 7s
44 actionable tasks: 44 executed
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Reply by frankcalise** [reply]
Consider scanning this issue, not sure if it is related: https://github.com/infinitered/ignite/issues/2431
